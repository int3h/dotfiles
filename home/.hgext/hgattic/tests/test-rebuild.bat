@echo off

rem setup hgattic in a temp test directory
setlocal
cd ..
cd > tmpfile
set /p atticdir= < tmpfile
erase tmpfile
cd tests
mkdir test
cd test
call hg init
cd .hg
echo [extensions] >tmp
echo hgattic=%atticdir%\attic.py >>tmp

rem now that attic shelves hg format patches all the time a user needs to be
rem configured for tests and datetime information must be provided for the
rem shelve command
echo [ui] >>tmp
echo username=Fred Widget ^<fred@example.com^>>>tmp
echo [defaults] >>tmp
echo attic-shelve = -d "0 0" >>tmp
move tmp hgrc
cd ..

rem replace here with test content
echo ### echo a ^> a.txt
echo a > a.txt
echo ### call hg addrem
call hg addrem
echo ### call hg st
call hg st
echo ### call hg shelve --git -q a
call hg shelve --git -q a
echo ### call hg unshelve -q
call hg unshelve -q
echo ### type .hg\attic\a
type .hg\attic\a
echo ### echo b ^>^> a.txt
echo b >> a.txt
echo ### call hg shelve --refresh
call hg shelve --refresh
echo ### type .hg\attic\a
type .hg\attic\a
echo ### call hg shelve -r -m test
call hg shelve -r -m test
echo ### type .hg\attic\a
type .hg\attic\a
echo ### call hg shelve -r b
call hg shelve -r b
echo ### call hg shelve -r -f b
call hg shelve -r -f b
rem end test content

rem cleanup
cd ..
rmdir /S /Q test
endlocal
