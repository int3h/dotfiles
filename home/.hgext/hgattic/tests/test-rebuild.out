 ### echo a > a.txt
### call hg addrem
adding a.txt
### call hg st
A a.txt
### call hg shelve --git -q a
### call hg unshelve -q
### type .hg\attic\a
# HG changeset patch
# User Fred Widget <fred@example.com>
# Date 0 0
diff --git a/a.txt b/a.txt
new file mode 100644
--- /dev/null
+++ b/a.txt
@@ -0,0 +1,1 @@
+a 
### echo b >> a.txt
### call hg shelve --refresh
patch a refreshed
### type .hg\attic\a
# HG changeset patch
# User Fred Widget <fred@example.com>
# Date 0 0
diff --git a/a.txt b/a.txt
new file mode 100644
--- /dev/null
+++ b/a.txt
@@ -0,0 +1,2 @@
+a 
+b 
### call hg shelve -r -m test
patch a refreshed
### type .hg\attic\a
# HG changeset patch
# User Fred Widget <fred@example.com>
# Date 0 0
test
diff --git a/a.txt b/a.txt
new file mode 100644
--- /dev/null
+++ b/a.txt
@@ -0,0 +1,2 @@
+a 
+b 
### call hg shelve -r b
abort: a different patch is active
### call hg shelve -r -f b
patch b refreshed
